teachertool/src/components/EditorContainer.tsx

```diff
@@ -0,0 +1,32 @@
+import { MakeCodeFrame } from "./MakecodeFrame";
+import { isLocal, getEditorUrl } from "../utils";
+
+const createIFrameUrl = (): string => {
+    const editorUrl: string = isLocal() ? "http://localhost:3232/index.html#editor" : getEditorUrl((window as any).pxtTargetBundle.appTheme.embedUrl);
+
+    let url = editorUrl
+    if (editorUrl.charAt(editorUrl.length - 1) === "/" && !isLocal()) {
+        url = editorUrl.substr(0, editorUrl.length - 1);
+    }
+    url += `?controller=1&ws=browser&nocookiebanner=1`;
+    return url;
+}
+
+export const EditorContainer: React.FC<{}> = () => {
```

Comment:
Not sure what value the `EditorContainer` container brings. Can this be rolled into `MakeCodeFrame`?

Reply:
The idea here is that there's something that wraps around the `MakeCodeFrame` for the teacher tool in particular to use, especially with the hope in mind that the `MakeCodeFrame` can be moved into react-common at some point. I'm under the impression that this is likely where we would handle the receiving blocks logic and things like that, as well. From playing with the skillmap iframe and thinking about generic cases, I imagined that the parent app should be able to control how the iframe is opened and closed, so the wrapper also provides a good place to define those functions.

The alternative would be to just put the functions and url-defining logic in the app base, but it felt a bit out of place.